### PR TITLE
fix(ci): fail-fast the Windows test-suite hang (8aef79e8 + ba388d01)

### PR DESCRIPTION
**Pull request by harshanandak**
## Summary

Makes the full local test suite actually complete on Windows instead of hanging up to the 15-min ceiling — removes the need for the `--quick-only` workaround.

Two related test-infra hangs:

- **8aef79e8** — pre-push full suite could block ~15 min. The default per-lane wall-clock ceiling (`DEFAULT_TEST_COMMAND_TIMEOUT_MS` in `scripts/test.js`) was 15 min. Lowered to **5 min** so a hung lane trips the ceiling and fails the push fast. Still overridable via `FORGE_TEST_TIMEOUT_MS`.
- **ba388d01** — `test/commands/remember.test.js` intermittently hung. The kernel broker's `git rev-parse --git-common-dir` probe (`lib/kernel/broker.js` `resolveGitCommonDir`, run on every broker init and exercised by remember/recall) had **no timeout**. Bounded it to **5s** so a Windows git hang fails fast (ETIMEDOUT) instead of wedging init. Also bounded the throwaway-repo `git init` in `test/helpers/kernel-project-root.js`.

## Changes
- `scripts/test.js` — default lane timeout 15 min → 5 min
- `lib/kernel/broker.js` — bounded `resolveGitCommonDir` git spawn (`GIT_COMMON_DIR_TIMEOUT_MS = 5000`)
- `test/helpers/kernel-project-root.js` — bounded helper `git init`
- `test/kernel/broker.test.js` — assert the new bounded exec option

## Verification
- `test/commands/remember.test.js` + `test/scripts/test-runner.test.js` + `test/kernel/broker.test.js`: **41 pass / 0 fail**
- Broader `test/kernel/` + `test/commands/` chunk: no hang
- `eslint --max-warnings 0` on all changed files: clean

Refs 8aef79e8, ba388d01.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Added a bounded timeout for Git “common dir” detection and improved resilience by falling back to the conventional `.git` location when detection fails, times out, or returns empty output (with a warning).

* **Tests**
  * Reduced the targeted test command timeout to 5 minutes and added separate full-suite timeout coverage (including `FORGE_TEST_TIMEOUT_MS` override behavior).
  * Updated Git-related mocks and added new assertions for graceful fallback and cleanup behavior during project root setup.

### CODE REVIEW

File: scripts/test.js [hunk 1]
```diff
@@ -44,8 +44,13 @@ const isWindows = process.platform === 'win32';
...
 // Windows, issue 8aef79e8). This kills the lane process so the push fails fast
-// instead of hanging forever. Override with FORGE_TEST_TIMEOUT_MS.
-const DEFAULT_TEST_COMMAND_TIMEOUT_MS = 15 * 60 * 1000;
+// instead of hanging forever.
+//
+// The default is a practical 5 minutes: a healthy full local suite completes
+// well under this, so a hang trips the ceiling in ~5 min instead of the old
+// 15-min wait that made `forge push` feel wedged. Raise it with
+// FORGE_TEST_TIMEOUT_MS for slow machines or intentionally long lanes.
+const DEFAULT_TEST_COMMAND_TIMEOUT_MS = 5 * 60 * 1000;
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply scripts/test.js:53]: ✅ Fixed: full-suite fallback now uses a validation-aligned budget.

Changed: scripts/test.js — added DEFAULT_FULL_SUITE_TIMEOUT_MS (10 min) + resolveFullSuiteTimeoutMs(). The full-suite fallback lane (scripts/test-full-suite.js) now runs with this larger budget; targeted/e2e/edge-case lanes keep the 5-min fail-fast ceiling. FORGE_TEST_TIMEOUT_MS still overrides both.
Reason: 10 min matches lib/commands/validate.js VALIDATION_COMMAND_TIMEOUT_MS=600000 and its 'long enough subprocess timeout for the full local suite' regression, so a healthy-but-slow full run (5-10 min) is no longer killed by the targeted-lane ceiling.
Coverage: test/scripts/test-runner.test.js — full-suite lane budget test + resolveFullSuiteTimeoutMs unit test.
Commit: 7cec4654

File: test/helpers/kernel-project-root.js [hunk 2]
```diff
@@ -28,7 +28,9 @@ function createKernelProjectRoots(prefix) {
 
   function makeProjectRoot() {
     const dir = fs.mkdtempSync(path.join(os.tmpdir(), prefix));
-    execFileSync('git', ['init', '-q'], { cwd: dir });
+    // Bounded timeout so a Windows git hang fails fast instead of stalling the
+    // suite until an outer ceiling kills it (issue ba388d01).
+    execFileSync('git', ['init', '-q'], { cwd: dir, timeout: 5000 });
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply test/helpers/kernel-project-root.js]: ✅ Fixed: reworded as best-effort + fixed a real temp-dir leak.

Changed: test/helpers/kernel-project-root.js.
1) Comment now states this is a best-effort timeout, not a hard 5s bound — execFileSync still waits for git to exit after the SIGTERM, so a genuinely hung git can exceed 5s; a strict ceiling would need a killable process-tree supervisor.
2) Stability fix: the mkdtemp'd dir is now pushed to tempDirs BEFORE 'git init' runs, so a failed/timed-out init no longer strands the temp dir — cleanup() always drains it.
Made execFileSync injectable to cover the failure path.
Coverage: new test/helpers/kernel-project-root.test.js — happy path, no-leak-on-git-init-failure, and bounded-timeout assertions.
Commit: 7cec4654
